docs/synapse/model_format.rst

```diff
@@ -0,0 +1,234 @@
+the following effects:
+
+    - On new Cortex's, the rev0 function will be loaded (adding the complete data model). Then subsequent functions
+      will be executed.  These should perform any model changes and data migrations as needed.
+    - On existing Cortexes which have already had the base model loaded; the rev0 function will be skipped and
+      subsequent functions executed until the last function has been executed.  These functions should perform a model
+      addition of the basemodel contents, and perform any required migration actions using the **storage layer** APIs.
+
+The following is a second revision of the earlier FooBarModule - it add a property for the foo:bar type, and performs
+a storage layer migration of nodes to set a default value for the new property. The model revision for the foobar model
+will be updated to 201707210101 in the cortex.  Note that during the model revision function, we exit the function
+early if the new property already exists.  This would cause the current model version to be set and any subsequent
+functions called.::
+
+    import synapse.lib.module as s_module
+
+    def getDataModel():
+        return FooBarModule.getBaseModels()[0][1]
+
+    class FooBarModule(s_module.CoreModule):
+
+        @staticmethod
+        def getBaseModels():
+            modl = {
+                'types': (
+                    ('foo:bar', {'subof': 'str', 'doc': 'A foo bar!'}),
+                ),
+                'forms': (
+                    ('foo:bar',
+                     {'ptype': 'str'},
+                     [('duck', {'defval': 'mallard', 'ptype': 'str', 'doc': 'Duck type.'})]
+                     ),
+                ),
+            }
+            name = 'foobar'
+            return ((name, modl), )
+
+        @s_module.modelrev('foobar', 201707210101)
+        def _testRev1(self):
+            '''
+            This revision adds the 'duck' property to our foo:bar nodes with its default value.
+            '''
+            try:
+                self.core.addPropDef('foo:bar:duck', form='foo:bar', defval='mallard', ptype='str', doc='Duck value!')
+            except DupPropName:
+                # The property already exists - return early
+                return
+            # Now lets migrate existing nodes to accommodate model changes.
+            rows = []
+            tick = s_common.now()
+            for iden, p, v, t in self.core.getRowsByProp('foo:bar'):
+                rows.append((iden, 'foo:bar:duck', 'mallard', tick))
+            self.core.addRows(rows)
+
+It is highly encouraged for model developers to write unit tests for any migrations which they do, in order to ensure
+that their migration functions are working correctly.
+
+Advanced CoreModule Usage
+-------------------------
+
+The CoreModule class can be used to define model handlers can be used to not just add models, but also add event
```

Comment:
Seem to be missing a word in line 131 - something like "...can be used to define model handlers AND can be used to not just add models..." (emphasis mine) ?